### PR TITLE
Add preview to Dance AI modal

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Adds a preview to the dance AI modal. Position/placement/size is of course subject to change as designs are finalized.

Example (note that the preview only renders the output of the AI block, not whatever else is in the workspace)

Screenshot 2023-10-05 at 4 31 22 PM

### CODE REVIEW

File: apps/src/dance/DanceAiModal.tsx [hunk 1]
```diff
@@ -192,19 +199,19 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
...
+    blocksSvg[1].nextConnection.connect(blocksSvg[2].previousConnection);
 
-    if (
-      origBlock &&
-      currentAiModalField &&
-      blocksSvg[0].previousConnection &&
-      blocksSvg[1].previousConnection &&
-      blocksSvg[2].previousConnection &&
-      blocksSvg[2].nextConnection
-    ) {
```
Comment (breville) [inline apps/src/dance/DanceAiModal.tsx]: Nice to see this gone; did we no longer get TypeScript complaints about these?

File: apps/src/dance/DanceAiModal.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import classNames from 'classnames';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
 import {AiModalItem, AiModalReturnedItem} from './types';
 import {BlockSvg} from 'blockly/core';
+import AiPreview from './ai/AiPreview';
```
Comment (breville) [inline apps/src/dance/DanceAiModal.tsx]: Nice.  An upcoming PR is going to move the other files to this directory too!

**Review by breville — APPROVED** [review]
Awesome!

File: apps/src/dance/ai/DanceAiModal.tsx [hunk 3]
```diff
@@ -139,10 +144,14 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
     setResponseExplanation(response.explanation);
   };
 
-  const convertBlocks = () => {
+  /**
+   * Generates blocks from the AI response in the main workspace, and attaches
+   * them to each other.
+   */
+  const generateBlocksFromResponse = (): [BlockSvg, BlockSvg, BlockSvg] => {
```
Comment (fisher-alice) [inline apps/src/dance/ai/DanceAiModal.tsx:151]: Nice! Thanks for breaking this out.

File: apps/src/dance/ai/DanceAiModal.tsx [hunk 4]
```diff
@@ -396,6 +405,7 @@ const DanceAiModal: React.FunctionComponent<DanceAiProps> = ({onClose}) => {
               onTypingDone={() => {
                 setTypingDone(true);
                 currentAiModalField?.setValue(responseJson);
+                setShowPreview(true);
```
Comment (fisher-alice) [inline apps/src/dance/ai/DanceAiModal.tsx:408]: This is probably more of a question for @samantha-code.  Now that we show the preview in the modal, should we wait to show the ai-generated background in the visualization column until modal is closed?
For example, if there are already background/foreground effects in the `setup` block, and then we click and drag a new ai block, I wonder if it's a little confusing to show the new generated background in both the modal and vis column before the user actually selects 'use' or 'convert'.

Comment (fisher-alice) [reply apps/src/dance/ai/DanceAiModal.tsx:408]: Current:


https://github.com/code-dot-org/code-dot-org/assets/107423305/d862e378-4c32-4717-97f8-92a1569e87a1



If we don't show till after modal is closed:

https://github.com/code-dot-org/code-dot-org/assets/107423305/a44b5319-be16-49fb-885b-fb0eb8757392

Comment (fisher-alice) [reply apps/src/dance/ai/DanceAiModal.tsx:408]: For after screencast, moved `currentAiModalField?.setValue(responseJson);` to `handleDoneClick`.

**Review by fisher-alice — COMMENTED** [review]
This looks great! Just left a product question for Sam.

File: apps/src/dance/ai/DanceAiModal.tsx (hunk 4 above)
Comment (fisher-alice) [reply apps/src/dance/ai/DanceAiModal.tsx:408]: Posted in [Slack thread](https://codedotorg.slack.com/archives/C05P12U42F5/p1696863070318059) and product/curric both wanted to show generated background after modal closed. I'll create a trello item for this follow-up task.

**Review by fisher-alice — APPROVED** [review]

**Pull request merged**